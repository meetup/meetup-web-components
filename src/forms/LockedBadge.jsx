import PropTypes from 'prop-types';
import React from 'react';

import Icon from '../media/Icon';
import Flex from '../layout/Flex';
import FlexItem from '../layout/FlexItem';

import DeprecationWarning from '../utils/components/DeprecationWarning';
/**
 * @module LockedBadge
 */
export class LockedBadge extends React.PureComponent {
	constructor(props) {
		super(props);

		this.variants = {
			default: {
				iconColor: '#008294',
				labelColor: '#008294',
				backgroundColor: '#dcebee',
			},
			neutral: {
				iconColor: '#000000',
				labelColor: '#212121',
				backgroundColor: '#f6f7f8',
			},
		};

		this.handleClick = this.handleClick.bind(this);
	}

	componentDidCatch(error, info) {
		console.log(`${error}: \n ${info.componentStack}`);
	}

	handleClick(e) {
		e.preventDefault();

		this.props.onClick(e);
	}

	render() {
		const {
			label,
			onClick, // eslint-disable-line no-unused-vars
			variant = 'default',
		} = this.props;

		const { iconColor, labelColor, backgroundColor } = this.variants[variant];

		return (
			<Flex>
				<FlexItem
					shrink
					onClick={this.handleClick}
					className="lockedBadge-container"
					style={{ backgroundColor }}
				>
					<div className="lockedBadge-badge">
						<Icon shape="lock" size="xs" color={iconColor} />
						<span className="lockedBadge-label" style={{ color: labelColor }}>
							{label}
						</span>
					</div>
				</FlexItem>
			</Flex>
		);
	}
}

LockedBadge.propTypes = {
	/** Style theme for badge */
	variant: PropTypes.oneOf(['default', 'neutral']),

	/** Label of CTA badge for unlocking the panel */
	label: PropTypes.string.isRequired,

	/** A callback that happens after the locked label has been clicked  */
	onClick: PropTypes.func.isRequired,
};

export default DeprecationWarning(LockedBadge);
