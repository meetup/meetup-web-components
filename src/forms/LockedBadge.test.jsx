import React from 'react';
import { shallow } from 'enzyme';
import LockedBadge from './LockedBadge';

describe('LockedBadge', function() {
	const onClickCallbackMock = jest.fn();

	const MOCK_PROPS = {
		label: 'Default label',
		onClick: onClickCallbackMock,
	};

	const render = (props = MOCK_PROPS) => {
		const wrapper = shallow(<LockedBadge {...props} />);

		return wrapper;
	};

	afterEach(() => {
		onClickCallbackMock.mockClear();
	});

	it('renders locked badge with default styles', () => {
		expect(render()).toMatchSnapshot();
	});
	it('calls onClickCallback when user clicks on badge', () => {
		const badge = render();
		expect(onClickCallbackMock).not.toHaveBeenCalled();
		badge.simulate('click');
		expect(onClickCallbackMock).toHaveBeenCalledTimes(1);
	});
});
