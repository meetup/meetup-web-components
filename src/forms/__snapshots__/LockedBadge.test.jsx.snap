// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LockedBadge renders locked badge with default styles 1`] = `
<LockedBadge
  label="Default label"
  onClick={[MockFunction]}
  preventDefault={true}
/>
`;

exports[`LockedBadge renders locked badge with neutral styles 1`] = `
<LockedBadge
  label="Default label"
  onClick={[MockFunction]}
  preventDefault={true}
  variant="neutral"
/>
`;
